src/components/NavBar/NavBar.tsx

```diff
@@ -36,14 +36,15 @@ const UserManagementNavItem: React.FC<NavBarProps> = (groups) => {
 
 const NavBar: React.FC<NavBarProps> = ({ groups }) => {
   const classes = useCustomStyles()
+  const isSupervisor = groups.some((group) => group === UserGroup.Supervisor)
```

Comment:
Could we extend `hasAccess` function on the user entity to determine this permission? There is an `isSupervisor` function defined as part of that logic